app/scripts/controllers/deployment.js

```diff
@@ -112,6 +124,7 @@ angular.module('openshiftConsole')
             $scope.loaded = true;
             $scope.deployment = deployment;
             updateHPAWarnings();
+            EnvironmentService.toggleReadonlyForContainerEnvsValueFrom(_.get($scope.deployment, 'spec.template.spec.containers'), envSecretAndConfigMapAreReadonly);
```

Comment:
We considered moving the `DataService.list("secrets")` and `DataService.list("configmaps")` into the directive & eliminating the controller noise... might be a good way to fix this.